### PR TITLE
Prepare NEWS entries for v1.4.0

### PR DESCRIPTION
**Pull request by kyleam**
Entries for https://github.com/metrumresearchgroup/bbr/compare/1.3.1...0df5954cb73e1d1fc61fd7bff8bfcf15a9a841d6

<details>
<summary>PR list</summary>

Candidates generated with `git log --first-parent --oneline --no-decorate 1.3.1..main`.

A NEWS entry was not added for the PRs marked with `-`.

```
  0df5954c Merge pull request #539 from metrumresearchgroup/patch/row-print
  3f3976d1 Merge pull request #530 from metrumresearchgroup/feat/nm_join-repair-names
  6b273109 Merge pull request #537 from metrumresearchgroup/print-args-update
- d9f298a2 Merge pull request #540 from metrumresearchgroup/ci-bbi-latest
- 901673af Merge pull request #541 from metrumresearchgroup/clean-up-test-print
  2a59eac0 Merge pull request #514 from metrumresearchgroup/patch/threads
  a5b56cf7 Merge pull request #533 from metrumresearchgroup/patch/dup-row
  865e4d51 Merge pull request #527 from metrumresearchgroup/summary-multi
  14779e35 Merge pull request #524 from metrumresearchgroup/patch/bbi_null_num
  3e559b44 Merge pull request #526 from metrumresearchgroup/feat/filter_run_log
- 2c5287f5 Merge pull request #534 from metrumresearchgroup/bbi-binary-update
- 088feb83 Merge pull request #528 from metrumresearchgroup/patch/update-MPN
- cd0395a6 Merge pull request #525 from metrumresearchgroup/roxygen2-update
  2bd7c5c5 Merge pull request #522 from metrumresearchgroup/model-diff-identical
  8da7c52e Merge pull request #515 from metrumresearchgroup/feat/omega_matrix
  03389943 Merge pull request #519 from metrumresearchgroup/patch/test-threads
  e90a5564 Merge pull request #506 from metrumresearchgroup/objfn-nan
  6f76c02e Merge pull request #510 from metrumresearchgroup/feat/file-extension
  f31d080e Merge pull request #511 from metrumresearchgroup/patch/check_run_times
  0032b76b Merge pull request #508 from metrumresearchgroup/fix/nm_join
- e389715d Merge pull request #503 from metrumresearchgroup/patch/bbi_tips_tricks
- 2b6923fc Merge pull request #507 from metrumresearchgroup/req-cleanup
- fa717055 Merge pull request #500 from metrumresearchgroup/fix/requirement_ids
- a0bb2003 Merge pull request #504 from metrumresearchgroup/readme-update-default-branch
  f49748b8 Merge pull request #496 from metrumresearchgroup/patch/test-threads-error
  2e407cc7 Merge pull request #499 from metrumresearchgroup/patch/use_bbi
  d1df3597 Merge pull request #497 from metrumresearchgroup/patch/param_estimates_pval
  85886890 Merge pull request #487 from metrumresearchgroup/patch/starred-models
  6f00ed98 Merge pull request #493 from metrumresearchgroup/validation/test_to_reqs
  5b57feed Merge pull request #489 from metrumresearchgroup/patch/test-threads
  d7e4191b Merge pull request #473 from metrumresearchgroup/feat/test_threads_helper
  1149a981 Merge pull request #484 from metrumresearchgroup/patch/filter-log
  dcc58350 Merge pull request #480 from metrumresearchgroup/feat/wait_for_nonmem
- 9bfd1867 Merge pull request #471 from metrumresearchgroup/main
```

</details>

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,10 +1,72 @@
...
+
+- If the caller specifies a `threads` value above one in `.bbi_args` but does
+  _not_ specify `parallel`, `parallel = TRUE` is added so that the `threads`
+  value is in effect.  (#514)
+
+- New functions `check_nonmem_finished()` and `wait_for_nonmem()` enable
+  checking and waiting on models submitted to the grid.  (#480)
+
+- Added `test_threads()` for benchmarking simulation run times with various
+  threads values.  (#473, #489, #519)
```
[the remark this answers is not in the document]
Comment (barrettk) [reply NEWS.md:33]: Not relevant to this code section, but wondering if we want to add this:
` - Updated parallel tips and tricks vignette to reference new test_threads() function and related helpers. (#503)`

File: NEWS.md [hunk 2]
```diff
@@ -1,10 +1,72 @@
...
+  speed up when many models are passed.  (#527)
+
+- `model_diff()` learned to print a message rather than call
+  `diffobj::diffFile()` when there are no changes to avoid confusingly
+  displaying the entire file.  (#522)
+
+- `use_bbi` now creates leading directories if needed.  (#499)
+
+- `new_model` now ignores the extension of the supplied path.  (#510)
+
```
Comment (barrettk) [inline NEWS.md:56]: How do you feel about adding `read_model()` to this, and also adding "valid" before extensions, given that not all extensions are technically ignored (just `.ctl` and `.mod` from what I remember. 

In my "fake news" branch I had (though an example might not be that important):
`new_model() and read_model() will now ignore valid extensions (new_model("1.ctl") works when new_model("1") was previously required). (#510)`

File: NEWS.md [hunk 3]
```diff
@@ -1,10 +1,72 @@
...
+- `param_estimates` gained an `.alpha` argument that's useful for identifying
+  the ETAs flagged by the `eta_pval_significant` heuristic.  (#497)
+
+- `nm_join()` now aborts if the join column has duplicate values, suggesting to
+  the caller that the table format probably needs to be widened to prevent
+  NONMEM from truncating the values.  (#533)
+
+- `nm_file()` and friends now detect duplicate column names and make them
+  unique.  (#530, #539)
```
Comment (barrettk) [inline NEWS.md:23]: Should this be classified as a bug fix? Im ok to leave it here, but previously `nm_file()` would error if there were duplicate column names in the input file

**Review by barrettk — CHANGES_REQUESTED** [review]
Looks good overall. One (possibly two) requests and a couple minor questions/concerns

File: NEWS.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyleam) [reply NEWS.md:33]: > Can you add #542 to the test_threads PRs too (the one we just merged)

Sure.

>  wondering if we want to add this:
> - Updated parallel tips and tricks

Yeah, I was on the fence and decided to leave it out, but it doesn't hurt to point to it.

File: NEWS.md (hunk 2 above)
Comment (barrettk) [reply NEWS.md:56]: Ahh I take back the "valid" part. Looked through that PR and it seems all extensions are allowed. Though we only have tests for `.mod` and `.ctl`, which is interesting given that technically `.blah` would seemingly work

File: NEWS.md (hunk 3 above)
Comment (kyleam) [reply NEWS.md:23]: I suppose it depends on whether it's actually a bad file being fed in that bbr is being nice dealing with or if it's an expected case that we should handle.

But I don't think it matters much one way or the other, so I'll move it to the bugs section.

File: NEWS.md (hunk 2 above)
Comment (kyleam) [reply NEWS.md:56]: > Though we only have tests for .mod and .ctl

There's really not much point in doing comprehensive testing of `fs::path_ext_remove`.

> How do you feel about adding read_model() to this

I don't think that'd be accurate, but I'll confirm that I'm not mistaken.

File: NEWS.md (hunk 3 above)
Comment (barrettk) [reply NEWS.md:23]: Yeah thats a fair point. Entirely up to you then

File: NEWS.md (hunk 2 above)
Comment (barrettk) [reply NEWS.md:56]: > I don't think that'd be accurate, but I'll confirm that I'm not mistaken.

Yeah its not accurate. I recall talking with RyanM about doing that on a zoom call (and swore I saw it being added), but guess it didnt make it through

**Review by barrettk — APPROVED** [review]
LGTM. Thanks

**Pull request merged**